dist/selleckt.js

```diff
@@ -264,214 +312,555 @@ var _ = (typeof window !== "undefined" ? window._ : typeof global !== "undefined
+
+            self.close();
+        }
+
+        function scrollItems(offset, absolute){
+            lockMousover = true;
+            if (absolute) {
+                itemslist.scrollTop = offset;
+            } else {
+                itemslist.scrollTop += offset;
+            }
+
+            _.delay(function(){
+                lockMousover = false;
+            }, 200);
+        }
+
+        $popup.on('mouseover', itemClass, function(e){
+            if (lockMousover) {
+                return;
+            }
+            highlightItem($(e.currentTarget));
+        });
+
+        $popup.on('click', itemClass, function(e){
+            e.preventDefault();
+
+            return selectItem($(e.currentTarget));
+        });
+
+        $popup.on('keydown', itemClass, function(e){
+            var whichKey = e.which,
+                $currentHighlightItem,
+                $theItems,
+                $itemToHighlight;
+
+            if(whichKey !== KEY_CODES.DOWN && whichKey !== KEY_CODES.UP &&
+                whichKey !== KEY_CODES.ENTER && whichKey !== KEY_CODES.ESC){
+                return;
+            }
+
+            e.preventDefault();
+            e.stopPropagation();
+            e.stopImmediatePropagation();
+
+            if(e.which === KEY_CODES.ESC){
+                return self.close();
+            }
+
+            $currentHighlightItem = getHighlightItem();
+
+            if(e.which === KEY_CODES.ENTER){
+                return selectItem($currentHighlightItem);
+            }
+
+            $theItems = $popup.find(itemClass);
+
+            if(whichKey === KEY_CODES.DOWN){
+                $itemToHighlight = $currentHighlightItem.nextAll(itemClass).first();
+
+                if (!$currentHighlightItem.length || !$itemToHighlight.length) {
+                    $itemToHighlight = $theItems.first();
+                    scrollItems(0, true);
+                } else if ($itemToHighlight.offset().top +
+                            $itemToHighlight.outerHeight() >
+                            $popup.offset().top +
+                            $popup.outerHeight()) {
+                    scrollItems($itemToHighlight.outerHeight());
+                }
+
+                return highlightItem($itemToHighlight);
+            }
+
+            if(whichKey === KEY_CODES.UP){
+                $itemToHighlight = $currentHighlightItem.prevAll(itemClass).first();
+
+                if(!$currentHighlightItem.length || !$itemToHighlight.length){
+                    $itemToHighlight = $theItems.last();
+                    scrollItems($itemToHighlight.offset().top, true);
+                } else if ($itemToHighlight.offset().top < $popup.offset().top + $itemToHighlight.outerHeight()) {
+                    scrollItems(-$itemToHighlight.outerHeight());
+                }
+
+                return highlightItem($itemToHighlight);
+            }
+        });
+
+        if(this.showSearch){
```

Comment:
Lots of conditionals strewn over the code for search, might be better to plonk it into a plugin at some point (not now though)